### PR TITLE
Fix parsing invalid href

### LINKED ISSUE #501: The path of a URI with an authority must start with a slash "/" or be empty
Not sure if this is an issue for Guzzle, but I'll start here since I am curious of your opinion on this.

When fetching this URL: https://kipli.com/fr/product/matelas-bebe-bio/ (https://oscarotero.com/embed/demo/index.php?url=https%3A%2F%2Fkipli.com%2Ffr%2Fproduct%2Fmatelas-bebe-bio%2F&settings=)

My app returns a warning: `The path of a URI with an authority must start with a slash "/" or be empty`. After some digging, I found that the source code of the site seems to be invalid. In particular the alternative languages:
```html
<link data-n-head="ssr" rel="alternate" hreflang="en_dk" href="undefined">
```
The locale for `en_dk` has `undefined` set for the `href`. And I believe this is the cause of the warning I am getting. Your demo seems to work fine but I have
```php
$foo = $embedInfo->languages
```
inside a try/catch which is then failing. Because `undefined` is not a valid URL/URI. 

My question is if you can/should handle this and determine if the `href` is valid?

### PR DESCRIPTION
**Pull request by grandeljay**
This is my attempt to fix #501

> In this case, I'd create a helper, so it can be used in other places. Something like:

@oscarotero I wasn't sure what exactly you mean, but maybe something like this?

### CODE REVIEW

File: src/Detectors/Languages.php [hunk 1]
```diff
@@ -26,4 +26,13 @@ public function detect(): array
 
         return $languages;
     }
+
+    private function isEmpty($value): boolval
```
Comment (oscarotero) [inline src/Detectors/Languages.php:30]: This function should be here https://github.com/oscarotero/Embed/blob/master/src/functions.php

The return type `boolval` doesn't exist. It should be `bool`.

File: src/Detectors/Languages.php [hunk 2]
```diff
@@ -17,7 +17,7 @@ public function detect(): array
             $language = $node->getAttribute('hreflang');
             $href = $node->getAttribute('href');
 
-            if (!$language || !$href) {
+            if (isEmpty()) {
```
Comment (oscarotero) [inline src/Detectors/Languages.php:20]: This function hasn't any argument

**Review by oscarotero — CHANGES_REQUESTED** [review]

**Comment by grandeljay** [comment]
Sorry you had to see that. This time I tested my changes.

I'm also wondering if it would make sense to use the splat operator so we could use 
```php
if (self::isEmpty($language, $href)) {
```
instead?

```php
private function isEmpty(string ...$values): bool
{
    $skipValues = array(
        'undefined',
    );

    foreach ($values as $value) {
        if (empty($value) || in_array($value, $skipValues)) {
            return true;
        }
    }

    return false;
}
```

**Comment by oscarotero** [comment]
> I'm also wondering if it would make sense to use the splat operator

Ok, but the arguments can be of any type, not only `string` (it must return `true` for `null`, `0`, `[]`, etc).

The `isEmpty` function shouldn't be a private method but a function that can be imported by other classes. The global helpers are located in this file: https://github.com/oscarotero/Embed/blob/master/src/functions.php

**Comment by grandeljay** [comment]
> Ok, but the arguments can be of any type, not only string (it must return true for null, 0, [], etc).

`getAttribute()` seems to return a nullable string. If this helper is just used for this method, I think requiring a `string` parameter makes sense.

If you want to make the helper more "future proof" by allowing all types, I don't think we can continue to use `empty()` since it
> Returns true if var does not exist or has a value that is empty or equal to zero, aka falsey, [...]. Otherwise returns false.

So `null` and `0` would be considered empty values which you do not seem to want. 

> The isEmpty function shouldn't be a private method but a function that can be imported by other classes. The global helpers are located in this file: https://github.com/oscarotero/Embed/blob/master/src/functions.php

Thanks! I'll change it!

**Comment by oscarotero** [comment]
Looks good to me. Thanks!

**Pull request merged**